File: ARCADE/Intro/Level 9_2: knapsackLight.cs
    using System;
     
    public class Test
    {
    	public static int knapsackLight(int value1, int weight1, int value2, int weight2, int maxW){
            if (weight1 + weight2 <= maxW) return value1 + value2;
            if (Math.Min(weight1, weight2) > maxW) return 0;
            if (weight1 <= maxW && (value1 >= value2 || weight2 > maxW)) return value1;
            return value2;
        }
    	public static void Main()
    	{
    		int value1 = 10, weight1 = 5, value2 = 6, weight2 = 4, maxW = 8;
    		Console.WriteLine(knapsackLight(value1, weight1, value2, weight2, maxW));
    	}
    }

File: ARCADE/Intro/Level 9_2: knapsackLight.pas
program knapsackLight;

uses math;

function knapsackLight(value1: integer; weight1: integer; value2: integer; 
					   weight2: integer; maxW: integer): integer;
var rs : integer;
begin
    rs := 0;
    if (weight1 + weight2 <= maxW) then rs := value1 + value2
    else if (Min(weight1, weight2) > maxW) then rs := 0
        else if ((weight1 <= maxW) and ((value1 >= value2) or (weight2 > maxW))) then rs := value1
            else rs := value2;
    knapsackLight := rs;
end;

var value1, weight1, value2, weight2, maxW: integer;
begin
	value1 := 10; weight1 := 5; value2 := 6; weight2 := 4; maxW := 8;
	writeln(knapsackLight(value1, weight1, value2, weight2, maxW));
	value1 := 10; weight1 := 5; value2 := 6; weight2 := 4; maxW := 9;
	writeln(knapsackLight(value1, weight1, value2, weight2, maxW));
end.
